src/Console/Command/GenerateExtensionCommand.php

```diff
@@ -10,116 +10,367 @@
+        $this->info('Generating extension skeleton for "'.$package.'"...');
 
         $replacements = [
             '{{namespace}}' => $namespace,
-            '{{escapedNamespace}}' => str_replace('\\', '\\\\', $namespace),
-            '{{name}}' => $name
+            '{{name}}' => $package,
         ];
 
         $this->copyStub($dir, $replacements);
 
         $manifest = [
-            'name' => $name,
-            'title' => $title,
+            'name' => $packageName,
             'description' => $description,
+            'type' => 'flarum-extension',
             'keywords' => [],
-            'version' => '0.1.0',
-            'author' => [
-                'name' => $authorName,
-                'email' => $authorEmail,
-                'homepage' => ''
-            ],
             'license' => $license,
+            'authors' => [
+                [
+                    'name' => $authorName,
+                    'email' => $authorEmail,
+                ],
+            ],
             'require' => [
-                'flarum' => '>='.Application::VERSION
+                'flarum/core' => '^'.Application::VERSION,
+            ],
+            'autoload' => [
+                'psr-4' => [
+                    $namespace.'\\' => 'src/',
+                ],
+            ],
+            'extra' => [
+                'flarum-extension' => [
+                    'title' => $title,
+                    'icon' => [
+                        'name' => '',
+                        'backgroundColor' => '',
+                        'color' => '',
+                    ],
+                ],
             ],
-            'icon' => [
-                'name' => '',
-                'backgroundColor' => '',
-                'color' => ''
-            ]
         ];
 
-        file_put_contents($dir.'/flarum.json', json_encode($manifest, JSON_PRETTY_PRINT));
+        file_put_contents($dir.'/composer.json', json_encode($manifest, JSON_PRETTY_PRINT));
+
+        if (! $this->input->getOption('nojs')) {
+            $this->info('Downloading JavaScript dependencies');
+
+            passthru('cd '.escapeshellarg($dir).'; cd js/forum; npm install; gulp default; cd ../admin; npm install; gulp default');
+        }
 
-        passthru("cd $dir; composer install; cd js/forum; npm install; gulp; cd ../admin; npm install; gulp");
+        $this->info('Updating Composer');
 
-        $this->info('Extension "'.$name.'" generated!');
+        if (! isset($composer['repositories'])) {
+            $composer['repositories'] = [
+                [
+                    'type' => 'path',
+                    'url' => 'workbench\/*\/',
+                ],
+            ];
+        }
+
+        $composer['require'][$packageName] = 'dev-master';
+
+        file_put_contents($composerPath, json_encode($composer, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+
+        passthru('cd '.escapeshellarg(public_path()).'; composer update');
```

Comment:
This should be more specific? Don't want to update all their packages if they don't want to.

`passthru('cd '.escapeshellarg(public_path()).'; composer update '.$packageName);`